### PR TITLE
RavenDB-22351 Sharing the instance of table stats in case of switching between compressed and uncompressed schema in the same transaction

### PR DESCRIPTION
**Pull request by arekpalinski**
### Issue link

https://issues.hibernatingrhinos.com/issue/RavenDB-22351/Error-during-import

### Additional description

Please see the comment - https://issues.hibernatingrhinos.com/issue/RavenDB-22351/Error-during-import#focus=Comments-67-1052799.0-0

### Type of change

- [x] Bug fix
- [ ] Regression bug fix
- [ ] Optimization
- [ ] New feature

### How risky is the change?

- [x] Low 
- [ ] Moderate 
- [ ] High
- [ ] Not relevant

### Backward compatibility

- [ ] Non breaking change
- [ ] Ensured. Please explain how has it been implemented?
- [ ] Breaking change
- [ ] Not relevant

### Is it platform specific issue?

- [ ] Yes. Please list the affected platforms.
- [x] No

### Documentation update

- [ ] This change requires a documentation update. Please mark the issue on YouTrack using `Documentation Required` tag.
- [x] No documentation update is needed 

### Testing by Contributor

- [ ] Tests have been added that prove the fix is effective or that the feature works
- [ ] Internal classes added to the test class (e.g. entity or index definition classes) have the lowest possible access modifier (preferable `private`) 
- [x] It has been verified by manual testing

### Testing by RavenDB QA team

- [ ] This change requires a special QA testing due to possible performance or resources usage implications (CPU, memory, IO). Please mark the issue on YouTrack using `QA Required` tag.
- [x] No special testing by RavenDB QA team is needed

### Is there any existing behavior change of other features due to this change?

- [ ] Yes. Please list the affected features/subsystems and provide appropriate explanation
- [x] No

### UI work

- [ ] It requires further work in the Studio. Please mark the issue on YouTrack using `Studio Required` tag.
- [x] No UI work is needed

### CODE REVIEW

**Comment by arekpalinski** [comment]
test this in debug please

**Review by ppekrol — APPROVED** [review]

File: src/Voron/Data/Tables/Table.cs [hunk 1]
```diff
@@ -108,24 +108,19 @@ private void OnDataMoved(long previousId, long newId, byte* data, int size, bool
...
             if (_tableTree == null)
                 throw new ArgumentNullException(nameof(tableTree), "Cannot open table " + Name);
 
-            var stats = (TableSchemaStats*)_tableTree.DirectRead(TableSchema.StatsSlice);
-            if (stats == null)
-                throw new InvalidDataException($"Cannot find stats value for table {name}");
-
-            NumberOfEntries = stats->NumberOfEntries;
-            _overflowPageCount = stats->OverflowPageCount;
             _tablePageAllocator = new NewPageAllocator(_tx.LowLevelTransaction, _tableTree);
```
Comment (maciejaszyk) [inline src/Voron/Data/Tables/Table.cs:124]: Should we read stats from disk as same in old ctr and put the values into `TableSchemaStatsReference`?

Comment (maciejaszyk) [reply src/Voron/Data/Tables/Table.cs:124]: Nvm, see you've moved it into open table, but maybe add comment here they have to be already filled up.
BTW: then I guess you have to do it here as well: [src/Raven.Server/Storage/Schema/Updates/Documents/40014/From40013.cs](https://github.com/ravendb/ravendb/pull/18682/files#diff-7c54d5cbf1cac545a263fd811d47fd8c2f5fa1feea29903c4c5c02a67eed125b)

File: src/Voron/Data/Tables/TableSchemaStatsReference.cs [hunk 2]
```diff
@@ -0,0 +1,10 @@
+﻿using System.Runtime.InteropServices;
```
Comment (ArieSLV) [inline src/Voron/Data/Tables/TableSchemaStatsReference.cs]: Using directive is not required by the code and can be safely removed

File: src/Voron/Data/Tables/Table.cs [hunk 3]
```diff
@@ -2164,7 +2156,7 @@ public TableReport GetReport(bool includeDetails, StorageReportGenerator generat
         {
             generatorInstance ??= new StorageReportGenerator(_tx.LowLevelTransaction);
 
-            var overflowSize = _overflowPageCount * Constants.Storage.PageSize;
```
Comment (ArieSLV) [inline src/Voron/Data/Tables/Table.cs:2167]: Perhaps we want to create a computed property inside `TableSchemaStatsReference`?

```csharp
public long OverflowSize => OverflowPageCount * Constants.Storage.PageSize;
```

File: src/Voron/Data/Tables/Table.cs (hunk 1 above)
Comment (arekpalinski) [reply src/Voron/Data/Tables/Table.cs:124]: No need. In `From40013` we create `Table` instance only to call `AssertValidFixedSizeTrees()`. We don't do any modifications there.

For reference: https://github.com/ravendb/ravendb/pull/9300

File: src/Voron/Data/Tables/TableSchemaStatsReference.cs (hunk 2 above)
Comment (arekpalinski) [reply src/Voron/Data/Tables/TableSchemaStatsReference.cs]: done

File: src/Voron/Data/Tables/Table.cs (hunk 3 above)
Comment (arekpalinski) [reply src/Voron/Data/Tables/Table.cs:2167]: There is just one call like that. Let's leave as is.

**Review by ppekrol — APPROVED** [review]

**Review by ArieSLV — APPROVED** [review]

**Pull request merged**